src/main/java/io/confluent/connect/jdbc/validation/JdbcSourceConnectorValidation.java

```diff
@@ -279,36 +312,133 @@ private void validateModeColumnRequirements() {
-        String msg = "Mode '" + mode + "' requires incrementing column configuration. "
+        String msg = "Incrementing column configuration must be provided. "
+            + "Provide either 'incrementing.column.name' or 'incrementing.column.mapping'.";
+        addConfigError(JdbcSourceConnectorConfig.MODE_CONFIG, msg);
+        return false;
+      }
+    } else if (JdbcSourceConnectorConfig.MODE_TIMESTAMP.equals(mode)) {
+      // Mode TIMESTAMP: Need timestamp config, should NOT have incrementing config
+      if (!hasLegacyTimestampConfig && !hasNewTimestampConfig) {
+        String msg = "Timestamp column configuration must be provided. "
+            + "Provide either 'timestamp.column.name' or 'timestamp.columns.mapping'.";
+        addConfigError(JdbcSourceConnectorConfig.MODE_CONFIG, msg);
+        return false;
+      }
+      if (hasLegacyIncrementingConfig || hasNewIncrementingConfig) {
+        String msg = "Incrementing column mappings should not be provided";
+        addConfigError(JdbcSourceConnectorConfig.MODE_CONFIG, msg);
+        return false;
+      }
+    } else if (JdbcSourceConnectorConfig.MODE_INCREMENTING.equals(mode)) {
+      // Mode INCREMENTING: Need incrementing config, should NOT have timestamp config
+      log.debug("MODE_INCREMENTING validation - hasLegacyTimestampConfig: {}, "
+          + "hasNewTimestampConfig: {}", hasLegacyTimestampConfig, hasNewTimestampConfig);
+      log.debug("timestampColumnsMapping: {}", timestampColumnsMapping);
+      
+      if (!hasLegacyIncrementingConfig && !hasNewIncrementingConfig) {
+        String msg = "Incrementing column configuration must be provided. "
             + "Provide either 'incrementing.column.name' or 'incrementing.column.mapping'.";
-        addConfigError(JdbcSourceConnectorConfig.INCREMENTING_COLUMN_NAME_CONFIG, msg);
-        addConfigError(JdbcSourceConnectorConfig.INCREMENTING_COLUMN_MAPPING_CONFIG, msg);
+        addConfigError(JdbcSourceConnectorConfig.MODE_CONFIG, msg);
+        return false;
+      }
+      if (hasLegacyTimestampConfig || hasNewTimestampConfig) {
+        log.debug("Adding error for conflicting timestamp config in MODE_INCREMENTING");
+        final String msg = "Timestamp column mappings should not be provided";
+        addConfigError(JdbcSourceConnectorConfig.MODE_CONFIG, msg);
+        return false;
       }
     }
+
+    return true;
   }
 
   /**
    * Helper method to add validation errors to config values.
    */
   private void addConfigError(String configName, String errorMessage) {
-    connectorConfigValidationResult.configValues().stream()
+    validationResult.configValues().stream()
         .filter(cv -> cv.name().equals(configName))
         .findFirst()
         .ifPresent(cv -> cv.addErrorMessage(errorMessage));
   }
+
+  // Helper methods for configuration access
+  
+  /**
+   * Get table whitelist configuration as a set.
+   */
+  private Set<String> getTableWhitelistSet() {
```

Comment:
All these functions are not needed. Move all this to ConnectorConfig class and use that method.